python/pyspark/context.py

```diff
@@ -180,7 +180,7 @@ def __init__(
         udf_profiler_cls: Type[UDFBasicProfiler] = UDFBasicProfiler,
         memory_profiler_cls: Type[MemoryProfiler] = MemoryProfiler,
     ):
-        if "SPARK_REMOTE" in os.environ and "SPARK_LOCAL_REMOTE" not in os.environ:
```

Comment:
what is the difference between `SPARK_REMOTE` and `SPARK_ENABLE_CONNECT_MODE ` ?

Reply:
The difference is that spark remote indicates one particular endpoint and at the same time makes PySpark use spark connect. 

With this change spark remote enables the spark connect mode but does not modify the environment variable when a new session is created. I will clarify this more in the commit message.